### PR TITLE
WMTI calculations

### LINKED ISSUE #37: Add WMTI
We should add WMTI to the pipeline.

### PR DESCRIPTION
**Pull request by TheJaeger**
This PR adds WMTI calculations based on the paper [White Matter Characterization with Diffusional Kurtosis Imaging](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3136876/) and [NYU's wmti_parameters code](https://github.com/NYU-DiffusionMRI/DESIGNER/blob/master/parameter_estimation/wmti_parameters.m). Also closes #37, which has been on hold for really long.

Tested to be working on my end, and histograms of AWF and EAS parameters seem feasible.  Take note that application is fundamentally different from the one in DKE, so comparing against it is inadvisable.

### CODE REVIEW

File: designer/fitting/dwipi.py [hunk 1]
```diff
@@ -5,13 +5,17 @@
 import os
 import random as rnd
 import warnings
+
```
Comment (jbteves) [inline designer/fitting/dwipi.py]: No need to add a blank line here

File: designer/fitting/dwipi.py [hunk 2]
```diff
@@ -5,13 +5,17 @@
...
 import cvxpy as cvx
 import nibabel as nib
 import numpy as np
 from joblib import Parallel, delayed
 from scipy.special import expit as sigmoid
+import scipy.linalg as sla
 from tqdm import tqdm
+
```
Comment (jbteves) [inline designer/fitting/dwipi.py]: Or here

File: designer/fitting/dwipi.py [hunk 3]
```diff
@@ -767,6 +771,126 @@ class object
...
+        -------
+        awf:        axonal water fraction
+        eas_ad:     extra-axonal space axial diffusivity
+        eas_rd:     extra-axonal space radial diffusivity
+        eas_tort:   extra-axonal space tortuosity
+        ias_ad:     intra-axonal space axial diffusivity
+        ias_rd:     intra-axonal space radial diffusivity
+        ias_tort:   intra-axonal space tortuosity
+        """
+        def wmtihelper(dt, dir, adc, akc, awf, adc2dt):
```
Comment (jbteves) [inline designer/fitting/dwipi.py]: It's usually not a great idea to have a function definition inside of a function definition. It's really on recommended for incredibly complex applications. Is there some reason you chose to do that?

File: designer/pydesigner.py [hunk 4]
```diff
@@ -718,6 +718,23 @@
             dp.writeNii(kfa, img.hdr, op.join(metricpath, 'kfa'))
             dp.writeNii(mkt, img.hdr, op.join(metricpath, 'mkt'))
             dp.writeNii(trace, img.hdr, op.join(metricpath, 'trace'))
+            if args.wmti:
```
Comment (jbteves) [inline designer/pydesigner.py]: Should we make a separate path for WMTI?

File: designer/fitting/dwipi.py (hunk 3 above)
Comment (TheJaeger) [reply designer/fitting/dwipi.py]: The only reason why this is inside a another function is to make it compatible with `joblib` so the parent function could parallelize the computations. While I can also do the same with this function outside, I don't want to add methods or functions that don't process an entire DWI. This function is used nowhere else and I don't want it to ever be visible to the end-user.

File: designer/pydesigner.py (hunk 4 above)
Comment (TheJaeger) [reply designer/pydesigner.py]: I think the `wmti-` prefix is good enough. This is a DKI-based metric after all that won't appear if users don't want it. We won't be enabling this by default.

File: designer/fitting/dwipi.py (hunk 1 above)
Comment (TheJaeger) [reply designer/fitting/dwipi.py]: > Extra blank lines should be deleted, and I'm concerned about a function definition in a function definition.

I have another PR coming in to clean up and remove unused functions. That PR will take care of whitespaces and formatting.

File: designer/fitting/dwipi.py (hunk 3 above)
Comment (jbteves) [reply designer/fitting/dwipi.py]: I see. I suppose that's a compelling reason.

**Pull request merged**